pkg/devfile/generator/generators.go

```diff
@@ -91,6 +98,70 @@ func GetContainers(devfileObj parser.DevfileObj, options common.DevfileOptions)
 	return containers, nil
 }
 
+// GetInitContainers gets the init container for every preStart devfile event
+func GetInitContainers(devfileObj parser.DevfileObj) ([]corev1.Container, error) {
```

Comment:
No need to call with options, events maybe sequential/parallel. In case they are sequential, event2 may depend on event1 so would need to return all the events irrespective of the components being used by the tools or not with a custom option